services/npm/npm-rating.service.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const Joi = require('joi')
+const { starRating } = require('../text-formatters')
+const { BaseJsonService } = require('..')
+
+// https://api-docs.npms.io/#api-Package-GetPackageInfo
+const responseSchema = Joi.object({
+  score: Joi.object({
+    final: Joi.number().required(),
+    detail: Joi.object({
+      maintenance: Joi.number().required(),
+      popularity: Joi.number().required(),
+      quality: Joi.number().required(),
+    }),
+  }),
+}).required()
+
+module.exports = class NpmRating extends BaseJsonService {
+  static category = 'rating'
+
+  static route = {
+    base: 'npm',
+    pattern:
+      'rating/:type(maintenance|popularity|quality)?/:scope(@.+)?/:packageName',
+  }
+
+  static examples = [
+    {
+      title: 'NPM Rating',
+      namedParams: { packageName: 'egg' },
+      staticPreview: this.render({ score: 0.9711 }),
+      keywords: ['node'],
+    },
+    {
+      title: 'NPM Rating (Popularity)',
+      namedParams: { type: 'popularity', packageName: '@vue/cli' },
```

Comment:
If we're going to have an example package that is scoped let's be sure to use the `scope` and `packageName` params properly :wink:

Reply:
little help please.

when I change to `namedParams: { type: 'popularity', scope: '@vue', packageName: 'cli' },`

then got error:

```bash
Error: In example for NpmRating at index 1, expected "scope" to match "@.+", but got "%40vue"
```

how could I unescape it?

Reply:
Try adding the `pattern` key to the example object

https://github.com/badges/shields/blob/de019c2e529b9b3d20be953f0539563581215112/services/npm/npm-version.service.js#L30-L35